### PR TITLE
Improve lading container builds

### PR DESCRIPTION
**Pull request by blt**
### What does this PR do?

This commit attempts to build lading on amd64 and arm64. I will also attempt improvements in build speed. Note we now only build the lading binary when building containers.

### Related issues

REF SMPTNG-151

### CODE REVIEW

**Review by GeorgeHahn — APPROVED** [review]

File: .github/workflows/container.yml [hunk 1]
```diff
@@ -8,7 +8,54 @@ env:
...
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        platform: [linux/arm64]
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - uses: actions/checkout@v2.3.5
```
Comment (scottopell) [inline .github/workflows/container.yml:21]: I noticed in another container build job that the git checkout is fetching all branches and tags.
I believe we could get by with a sparse checkout https://github.com/actions/checkout?tab=readme-ov-file#fetch-only-the-root-files

Comment (blt) [reply .github/workflows/container.yml:21]: Oh I bet you're right! I'll fix that here.

Comment (blt) [reply .github/workflows/container.yml:21]: Addressed in [a84f3fd](https://github.com/DataDog/lading/pull/842/commits/a84f3fde270b69b5e2e132839429e760c76be648).

Comment (scottopell) [reply .github/workflows/container.yml:21]: Not quite, gotta bump the version too , `sparse-checkout` didn't exist in v2

Comment (blt) [reply .github/workflows/container.yml]: Addressed in [1683c68](https://github.com/DataDog/lading/pull/842/commits/1683c68c5b8ebb8041013c53d71dba7fe285480f) I do believe.

**Pull request merged**